hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java

```diff
@@ -367,8 +368,9 @@ protected FileSplit makeSplit(String segmentId, String filePath, long start, lon
       String[] deleteDeltaFilePath = null;
       if (isIUDTable) {
         // In case IUD is not performed in this table avoid searching for
-        // invalidated blocks.
-        if (CarbonUtil
+        // invalidated blocks. No need to check validation for splits written by SDK.
+        String segmentId = getSegmentIdFromFilePath(inputSplit.getFilePath());
```

Comment:
add more comments explaining SDK case segment id will always be null and using that as a decision point to skip this validation for SDK and why you need to skip it.

Reply:
Done